src/main/java/org/javawebstack/abstractdata/bson/BsonTypeAdapter.java

```diff
@@ -0,0 +1,88 @@
+package org.javawebstack.abstractdata.bson;
+
+import org.bson.*;
+import org.bson.codecs.BsonDocumentCodec;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.EncoderContext;
+import org.bson.io.BasicOutputBuffer;
+import org.bson.types.Decimal128;
+import org.bson.types.ObjectId;
+import org.javawebstack.abstractdata.AbstractElement;
+import org.javawebstack.abstractdata.mapper.MapperContext;
+import org.javawebstack.abstractdata.mapper.MapperTypeAdapter;
+import org.javawebstack.abstractdata.mapper.exception.MapperException;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public class BsonTypeAdapter implements MapperTypeAdapter {
+
+    private final BsonConverter converter;
+
+    public BsonTypeAdapter() {
+        this(new BsonConverter());
+    }
+
+    public BsonTypeAdapter(BsonConverter converter) {
+        this.converter = converter;
+    }
+
+    public AbstractElement toAbstract(MapperContext context, Object value) throws MapperException {
+        if(value instanceof byte[]) {
+            ByteBuffer buffer = ByteBuffer.wrap((byte[]) value);
+            buffer.order(ByteOrder.LITTLE_ENDIAN);
```

Comment:
Is this defined by the spec or should it be changable to `BIG_ENDIAN` also?

Reply:
This is defined by spec (see: https://bsonspec.org/spec.html) and i derived this logic from the bson internals. Unfortunately the bson library doesn't seem to provide any documented and easy to use api to actually convert the document to bson so we need to use internal api's here.